src/routes/transaction/transaction.js

```diff
@@ -0,0 +1,128 @@
+                                        currencyId: exchangeCurrency._id,
+                                        amount: 0
+                                    }}})
+        //get the index which is the last in array
+        exchangeCurrencyIndex = wallet.currency.length
+    }
+    //get updated wallet
+    wallet = await Wallet.findOne({ userId: req.session.passport.user._id })
+
+    await Wallet.bulkWrite([
+        {
+            updateOne: {
+                "filter": {_id: wallet._id, "currency.currencyId": baseCurrency._id },
+                "update": { $set: {"currency.$.amount": wallet.currency[baseCurrencyIndex].amount - amount} }
+            }
+        },
+        {
+            updateOne: {
+                "filter": {_id: wallet._id, "currency.currencyId": exchangeCurrency._id },
+                "update": { $set: {"currency.$.amount": wallet.currency[exchangeCurrencyIndex].amount + boughtCurrencyAmount} }
+            }
+        }
+    ])
+    await Currency.bulkWrite([
+        {
+            updateOne: {
+                "filter": {_id: baseCurrency._id},
+                "update": { $set: { exchangeAmount: baseCurrency.exchangeAmount - amount} }
+            }
+        },
+        {
+            updateOne: {
+                "filter": {_id: exchangeCurrency._id},
+                "update": { $set: { exchangeAmount: exchangeCurrency.exchangeAmount - boughtCurrencyAmount} }
+            }
+        }
+    ])
+    
+    let cryptoInWallet
+    if(baseCurrency._id.toString() !== xUSD._id.toString())
+    {
+        cryptoInWallet = wallet.currency[baseCurrencyIndex].amount - amount
+        //update ratio
+        await Currency.findByIdAndUpdate(
+            baseCurrency._id,
+            {
+                $set: { ratio: Math.random(0, 1)},
+            }
+        )
+
+        if(exchangeCurrency._id.toString() !== xUSD._id.toString())
+            //update ratio
+            await Currency.findByIdAndUpdate(
+                exchangeCurrency._id,
+                {
+                    $set: { ratio: Math.random(0, 1)},
+                }
+        )
+            
+    }  
+    else {
+        cryptoInWallet = wallet.currency[exchangeCurrencyIndex].amount + boughtCurrencyAmount
+        //update ratio
+        await Currency.findByIdAndUpdate(
+            exchangeCurrency._id,
+            {
+                $set: { ratio: Math.random(0, 1)},
+            }
+        )
+    }
+    let currencyInWallet = await Wallet.findOne({ userId: req.session.passport.user._id })
+
+    const transaction = await Transaction.create({
+        userId: req.session.passport.user._id,
+        baseId: baseCurrency._id,
+        exchangeId: exchangeCurrency._id,
+        soldAmount: amount,
+        boughAmount: boughtCurrencyAmount,
+        cryptoInWallet,
+        currencyInWallet: currencyInWallet.currency[xUsdIndexWallet].amount,
+        transactionDate: new Date()
+    })
+
+    res.status(200).send({
```

Comment:
Since you are returning early in the function, to be consitent, you should also return here at the end of the function

```suggestion
    return res.status(200).send({
```